src/test/java/com/baidu/hugegraph/api/VertexApiTest.java

```diff
@@ -209,6 +214,92 @@ public void testBatchCreate() {
         }
     }
 
+    @Test
+    public void testBatchUpdateStrategySum() {
+        BatchVertexRequest req = batchVertexRequest("price", 1, -1,
+                                                    UpdateStrategy.SUM);
+
+        this.graph().updateVertices(req).forEach(vertex -> {
+            Object price = vertex.properties().get("price");
+            Assert.assertTrue(price instanceof Number);
+            Assert.assertEquals(0, price);
+        });
+    }
+
+    @Test
+    public void testBatchUpdateStrategyBigger() {
+        BatchVertexRequest req = batchVertexRequest("price", 1, -1,
+                                                    UpdateStrategy.BIGGER);
+
+        this.graph().updateVertices(req).forEach(vertex -> {
+            Object price = vertex.properties().get("price");
+            Assert.assertTrue(price instanceof Number);
+            Assert.assertTrue((int) price > 0);
+        });
+    }
+
+    @Test
+    public void testBatchUpdateStrategySmaller() {
+        // TODO: Add date comparison after fixing the date serialization bug
+        BatchVertexRequest req = batchVertexRequest("price", -1, 1,
+                                                    UpdateStrategy.SMALLER);
+
+        this.graph().updateVertices(req).forEach(vertex -> {
+            Object price = vertex.properties().get("price");
+            Assert.assertTrue(price instanceof Number);
+            Assert.assertTrue((int) price < 0);
+        });
+    }
+
+    @Test
+    public void testBatchUpdateStrategyUnion() {
+        BatchVertexRequest req = batchVertexRequest("set", "old", "new",
+                                                    UpdateStrategy.UNION);
+
+        // TODO: List from server is unordered, consider better way to validate
+        this.graph().updateVertices(req).forEach(vertex -> {
+            Object set = vertex.properties().get("set");
+            Assert.assertTrue(set instanceof Collection);
+            Assert.assertEquals(2, ((Collection) set).size());
```

Comment:
add` Assert.assertEquals(ImmutableSet.of("xx", "xx"), set);`
same as other tests

Reply:
(Collection<?>) set